libs/external-data/src/readability/readability.service.ts

```diff
@@ -0,0 +1,220 @@
+
+      // calculate points for number of words per paragraph
+      let ppoints: number;
+      if (pratio <= 30) ppoints = 20;
+      else if (pratio >= 80) ppoints = 0;
+      else ppoints = this.toFixedNumber(20 - (pratio - 30) * 0.4, 2);
+
+      // add all points
+      const total_score = fkpoints + hpoints + ppoints;
+
+      const words = await this.removeStopwords(postText, lang);
+      const wordCount = await this.wordCount(words);
+
+      const dataWord =
+        Object.entries(wordCount)
+          .sort(([wordA, countA], [wordB, countB]) => countB - countA)
+          .slice(0, 20)
+          .map(([word, count]) => ({ word, count })) || [];
+
+      const date = new Date();
+
+      const dataObj = {
+        _id: new Types.ObjectId(),
+        original_score: original_fk,
+        final_fk_score: final_fk,
+        fkpoints: fkpoints,
+        pratio: pratio,
+        hratio: hratio,
+        len_par: len_par,
+        len_headings: len_headings,
+        total_score: total_score,
+        ppoints: ppoints,
+        hpoints: hpoints,
+        total_words: word,
+        syllables: syllables,
+        data_word: dataWord,
+        total_sentences: sentence,
+        total_syllables: syllables,
+      };
+
+      await this.db.collections.readability.insertMany({
+        ...dataObj,
+        url,
+        date,
+      });
+      wait(500);
+    }
+  }
+
+  async removeStopwords(text: string, lang: string): Promise<string[]> {
+    const words = text
+      .toLowerCase()
+      .replace(/[^\w\s]|_/g, '')
+      .replace(/\s+/g, ' ')
+      .split(' ');
+    const stopwords = new Set(lang === 'fr' ? stopwordsFr : stopwordsEn);
+    return words.filter((word) => !stopwords.has(word));
+  }
+
+  async wordCount(words: string[]): Promise<{ [key: string]: number }> {
+    const wordCount = {};
+    for (const word of words) {
+      wordCount[word] = (wordCount[word] || 0) + 1;
+    }
+    return wordCount;
+  }
+
+  toFixedNumber(num: number, digits: number, base = 10) {
+    const pow = Math.pow(base, digits);
+    return Math.round(num * pow) / pow;
+  }
+
+  syllableCountFr(text: string): number {
+    const words = text
+      .toLocaleLowerCase('fr-CA')
+      .replace(/l'|d'|j'|t'|s'|c'|n'|m'|l’|d’|j’|t’|s’|c’|n’|m’/g, '')
+      .replace(/[^A-zÀ-ú\-'’\s]|_/g, '')
+      .replace(/’/g, "'")
+      .replace(/\s+/g, ' ')
+      .split(' ');
+    const notFound = [];
+
+    const syllableCount = words.reduce((count, word) => {
+      const entry = lexiqueLookup[word];
+      if (entry !== undefined) {
+        return count + entry;
+      } else {
+        notFound.push(word);
+        return count;
+      }
+    }, 0);
+
+    return syllableCount;
+  }
+
+  fleschKincaidGradeFr(text: string): number {
+    const word = readability.lexiconCount(text);
+    const sentence = this.sentenceCount(text);
+    const syllables = this.syllableCountFr(text);
+    const asl = word / sentence;
+    const asw = syllables / word;
```

Comment:
```suggestion
    const syllablesPerWord = syllables / word;
```
Oh I think it means "average syllables per word" 😅

Still, any amount of ambiguity makes it that much harder to scan/read code. Try as much as possible to name variables in a way that it should be immediately clear what it is, even if you know nothing about the context. I'd rather be reading and typing long variable names than have to dig through code to figure out what something means.